encoding.bs

```diff
@@ -1347,20 +1350,108 @@ must run these steps:
    <li><p>Let <var>token</var> be the result of
    <a>reading</a> from <var>input</var>.
 
-   <li><p>Let <var>result</var> be the result of
-   <a>processing</a> <var>token</var> for
-   <a for=TextEncoder>encoder</a>, <var>input</var>, <var>output</var>.
+   <li><p>Let <var>result</var> be the result of <a>processing</a> <var>token</var> for the
+   <a>UTF-8 encoder</a>, <var>input</var>, <var>output</var>.
+
+   <li>
+    <p>Assert: <var>result</var> is not <a>error</a>.
+
+    <p class=note>The <a>UTF-8 encoder</a> cannot return <a>error</a>.
+
+   <li><p>If <var>result</var> is <a>finished</a>, convert <var>output</var> into a byte sequence,
+   and then return a {{Uint8Array}} object wrapping an {{ArrayBuffer}} containing <var>output</var>.
+   <!-- XXX https://www.w3.org/Bugs/Public/show_bug.cgi?id=26966 -->
+  </ol>
+</ol>
+
+<p>The
+<dfn method for=TextEncoder><code>encodeInto(<var>source</var>, <var>destination</var>)</code></dfn>
+method, when invoked, must run these steps:
+
+<ol>
+ <li><p>Let <var>read</var> be 0.
+
+ <li><p>Let <var>written</var> be 0.
+
+ <li><p>Let <var>destinationBytes</var> be the result of
+ <a lt="get a reference to the buffer source">getting a reference to the bytes held by</a>
+ <var>destination</var>.
+
+ <li>
+  <p>Let <var>unused</var> be a new <a for=/>stream</a>.
+
+  <p class=note>The <a>handler</a> algorithm invoked below requires this argument, but it is not
+  used by the <a>UTF-8 encoder</a>.
+
+ <li><p>Convert <var>source</var> to a <a for=/>stream</a>.
+
+ <li>
+  <p>While true:
+
+  <ol>
+   <li><p>Let <var>token</var> be the result of <a>reading</a> from <var>source</var>.
+
+   <li><p>Let <var>result</var> be the result of running the <a>UTF-8 encoder</a>'s <a>handler</a>
+   on <var>unused</var> and <var>token</var>.
+
+   <li><p>If <var>result</var> is <a>finished</a>, then <a for=iteration>break</a>.
 
    <li>
-    <p>If <var>result</var> is <a>finished</a>, convert <var>output</var> into a
-    byte sequence, and then return a {{Uint8Array}} object wrapping an
-    {{ArrayBuffer}} containing <var>output</var>.
-    <!-- XXX https://www.w3.org/Bugs/Public/show_bug.cgi?id=26966 -->
+    <p>Otherwise:
 
-    <p class=note><a>UTF-8</a> cannot return <a>error</a>.
+    <ol>
+     <li>
+      <p>If <var>destinationBytes</var>'s <a for="byte sequence">length</a> &minus;
+      <var>written</var> is greater than the number of bytes in <var>result</var>, then:
```

Comment:
I think this needs to be "greater than or equal to".

Reply:
Yeah, indeed, thanks! (E.g., if we have space for 12 bytes, already written 6 bytes, then writing 6 more bytes should be fine whereas the current wording prohibits that.)